products/extensions/search/How-to-modify-search-logic.md

```diff
@@ -0,0 +1,65 @@
+# How to modify search logic
+
+@See: `\Shopware\Commercial\AdvancedSearch\Domain\Search\SearchLogic`
+
+This class is the central place to build the Elasticsearch query:
+
+- Load all searchable fields of the wanted search entity and the current context's sales channel.
+- The search term will be tokenized and filtered into a list of "token". For e.g., `The 2 QUICK Brown-Foxes jumped over the lazy dog's bone.` will be tokenized to `[ The, 2, QUICK, Brown, Foxes, jumped, over, the, lazy, dog's, bone ]`.
+- Each search token will form a bool query to check whether the token is matched with any of the loaded searchable fields. This step is when `\Shopware\Commercial\AdvancedSearch\Domain\Search\TokenQueryBuilder::build` take place, it will help to build a `token query`.
+- These built queries will be combined into a single query by `AND` or `OR` operators, depending on the search behavior configured at the first step.
+- This query will be used by `\Shopware\Elasticsearch\Framework\DataAbstractionLayer\ElasticsearchEntitySearcher` to search
```

Comment:
```suggestion
* Load all searchable fields of the wanted search entity and the current context's sales channel.
* The search term will be tokenized and filtered into a list of "token". For e.g., `The 2 QUICK Brown-Foxes jumped over the lazy dog's bone` will be tokenized to `[ The, 2, QUICK, Brown, Foxes, jumped, over, the, lazy, dog's, bone ]`.
* Each search token will form a bool query to check whether the token matches any of the loaded searchable fields. This step is when `\Shopware\Commercial\AdvancedSearch\Domain\Search\TokenQueryBuilder::build` takes place, it will help to build a `token query`.
* These built queries will be combined into a single query by `AND` or `OR` operators, depending on the search behavior configured at the first step.
* This query will be used by `\Shopware\Elasticsearch\Framework\DataAbstractionLayer\ElasticsearchEntitySearcher` to search.
```